### PR TITLE
Metacal catalog for run2.1i coaddv1

### PR DESCRIPTION
**Pull request by EiffL**
Adds readers for metacal catalog coaddv1.
NOTE: I wasn't able  to move the parquet files to the  dpdd folder at NERSC, if someone can do it for me, or let me know how to do it, that would be a better location than my  scratch space

@joezuntz  You should be able  to use the metacal catalog from this  branch:
```
import sys
sys.path.insert(0, '/global/homes/f/flanusse/repo/gcr-catalogs/') # You need this branch: https://github.com/LSSTDESC/gcr-catalogs/tree/u/EiffL/metacal_composite
import GCRCatalogs

catalog = GCRCatalogs.load_catalog('dc2_object_run2.1i_dr1_with_metacal_griz')
```

### CODE REVIEW

File: GCRCatalogs/catalog_configs/dc2_metacal_griz_run2.1i_dr1a.yaml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+subclass_name: dc2_metacal.DC2MetacalCatalog
+base_dir: /global/homes/f/flanusse/scratch/object_mcal_catalog_coaddv1
+schema_filename: schema.yaml
+filename_pattern: 'metacal_tract_\d+\.parquet$'
+description: DC2 Run 2.1i coadd-v1 Metacal Catalog (griz)
+creators: ['Francois Lanusse', 'Johann Cohen-Tanugi']
+included_by_default: true
```
Comment (yymao) [inline GCRCatalogs/catalog_configs/dc2_metacal_griz_run2.1i_dr1a.yaml]:
```suggestion
```

File: GCRCatalogs/catalog_configs/dc2_metacal_griz_run2.1i_dr1a_tract4850.yaml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+subclass_name: dc2_metacal.DC2MetacalCatalog
+base_dir: /global/homes/f/flanusse/scratch/object_mcal_catalog_coaddv1
+schema_filename: schema.yaml
+filename_pattern: 'metacal_tract_4850.parquet$'
+description: DC2 Run 2.1i coadd v1 Metacal Catalog (griz)
+creators: ['Francois Lanusse', 'Johann Cohen-Tanugi']
+included_by_default: true
```
Comment (yymao) [inline GCRCatalogs/catalog_configs/dc2_metacal_griz_run2.1i_dr1a_tract4850.yaml]:
```suggestion
```

File: GCRCatalogs/catalog_configs/dc2_object_run1.2i_with_metacal.yaml [hunk 3]
```diff
@@ -0,0 +1,8 @@
+subclass_name: composite.CompositeReader
+description: DC2 Run 1.2i Object Catalog with metacal
+catalogs:
+  - catalog_name: dc2_metacal_run1.2i_tract5063
+    matching_method: 'id'
+  - catalog_name: dc2_object_run1.2i
+    matching_method: 'id'
+included_by_default: true
```
Comment (yymao) [inline GCRCatalogs/catalog_configs/dc2_object_run1.2i_with_metacal.yaml]:
```suggestion
include_in_default_list: true
```

File: GCRCatalogs/catalog_configs/dc2_object_run1.2i_with_metacal.yaml [hunk 4]
```diff
@@ -0,0 +1,8 @@
+subclass_name: composite.CompositeReader
+description: DC2 Run 1.2i Object Catalog with metacal
+catalogs:
+  - catalog_name: dc2_metacal_run1.2i_tract5063
+    matching_method: 'id'
+  - catalog_name: dc2_object_run1.2i
```
Comment (yymao) [inline GCRCatalogs/catalog_configs/dc2_object_run1.2i_with_metacal.yaml]:
```suggestion
  - catalog_name: dc2_object_run1.2i_tract5063
```

**Comment by yymao** [comment]
Also pinging @heather999 about the file copy issue.

**Comment by heather999** [comment]
@EiffL Just point me at what in your scratch area needs to be moved and to where.  I'm assuming this is going into `/global/projecta/projectdirs/lsst/production/DC2_ImSim/Run2.1i/dpdd`?

**Comment by EiffL** [comment]
Thanks @heather999 , the files to be moved are currently in `/global/homes/f/flanusse/scratch/object_mcal_catalog_coaddv1`

I think we want them in `/global/projecta/projectdirs/lsst/production/DC2_ImSim/Run2.1i/dpdd/calexp-v1\:coadd-v1/metacal_griz_catalog`

That is, if I understand the naming scheme  for the dpdd folder correctly. Let me know if you have any issues with accessing the files.

**Comment by joezuntz** [comment]
@EiffL could the main new catalog match with dc2_object_run1.2i_all_columns instead of dc2_object_run1.2i? There are some columns in the former that are useful in TXPipe.

**Comment by yymao** [comment]
@joezuntz before @EiffL fixes it, you should be able to use the following hack to get what you need.

```python
from GCRCatalogs.composite import CompositeReader

catalogs_config = [
    {'catalog_name': 'dc2_object_run2.1i_dr1a_all_columns', 'matching_method': 'id', 'filename_pattern': r'object_tract_4850\.hdf5$'},
    {'catalog_name': 'dc2_metacal_griz_run2.1i_dr1a_tract4850', 'matching_method': 'id'},
]

catalog = CompositeReader(catalogs=catalogs_config)
```

**Comment by joezuntz** [comment]
@yymao Thanks!  Would it be possible to add a GCRCatalog function to do this automatically from two existing catalogs?

**Comment by yymao** [comment]
@joezuntz Probably yes, but I need to know what "this" is that you were referring to :slightly_smiling_face:

File: GCRCatalogs/catalog_configs/dc2_object_run2.1i_dr1_with_metacal_griz.yaml [hunk 5]
```diff
@@ -0,0 +1,8 @@
+subclass_name: composite.CompositeReader
+description: DC2 Run 2.1i Object Catalog with metacal (griz)
+catalogs:
+  - catalog_name: dc2_object_run2.1i_dr1a
```
Comment (yymao) [inline GCRCatalogs/catalog_configs/dc2_object_run2.1i_dr1_with_metacal_griz.yaml]:
```suggestion
  - catalog_name: dc2_object_run2.1i_dr1a_all_columns
```

File: GCRCatalogs/catalog_configs/dc2_object_run2.1i_dr1_tract4850_with_metacal_griz.yaml [hunk 6]
```diff
@@ -0,0 +1,8 @@
+subclass_name: composite.CompositeReader
+description: DC2 Run 2.1i Object Catalog with metacal (griz)
+catalogs:
+  - catalog_name: dc2_object_run2.1i_dr1a_tract4850
```
Comment (yymao) [inline GCRCatalogs/catalog_configs/dc2_object_run2.1i_dr1_tract4850_with_metacal_griz.yaml]:
```suggestion
  - catalog_name: dc2_object_run2.1i_dr1a_all_columns
    filename_pattern: 'object_tract_4850\.hdf5$'
```

**Comment by EiffL** [comment]
@yymao  This should all work now, I'll make another PR for the  dr1b coadd

**Comment by EiffL** [comment]
yeah... hum... we need someone to copy over these files from `/global/homes/f/flanusse/scratch/object_mcal_catalog_coaddv1` to `/global/projecta/projectdirs/lsst/production/DC2_ImSim/Run2.1i/dpdd/calexp-v1\:coadd-v1/metacal_griz_catalog` @wmwv @heather999, can any of you do  it ?

**Comment by heather999** [comment]
Done, just let me know if there are any problems accessing the files.. the area should be readable by all LSST users:
```
desc@cori12:/global/projecta/projectdirs/lsst/production/DC2_ImSim/Run2.1i/dpdd/calexp-v1:coadd-v1> ls
./  ../  metacal_griz_catalog/	object_table_summary/
```

**Comment by EiffL** [comment]
Thanks @heather999 ! I changed the paths and tested the catalog, seems to work @yymao !

**Comment by yymao** [comment]
@EiffL I should have caught this earlier but 

- `dc2_object_run2.1i_dr1_tract4850_with_metacal_griz` should be renamed as `dc2_object_run2.1i_dr1a_tract4850_with_metacal_griz`

and 

- `dc2_object_run2.1i_dr1_with_metacal_griz` should be renamed as `dc2_object_run2.1i_dr1a_with_metacal_griz`

especially because `dc2_object_run2.1i_dr1` is already pointing to `dc2_object_run2.1i_dr1b`.

**Comment by yymao** [comment]
BTW, @EiffL, should v0.13.0 (which we want to tag tomorrow) wait for dr1b metalcal?

**Comment by EiffL** [comment]
Let me  rename these files, also, I just finished producing the parquet files for dr1b, might as well include them. It should only take an hour

**Comment by EiffL** [comment]
Let's merge this one first. There is a super weird issue with dr1b that I can't seem to figure out. I have placed the dr1b files in `u/EiffL/metacal_dr1b`

**Pull request merged**